src/cascadia/TerminalControl/ControlInteractivity.cpp

```diff
@@ -0,0 +1,560 @@
+                // create a positive delta, but we want the viewport to move up,
+                // so we'll need a negative scroll amount (and the inverse for
+                // panning down)
+                const float numRows = -1.0f * (dy / fontSizeInDips.height<float>());
+
+                const auto currentOffset = ::base::ClampedNumeric<double>(_core->ScrollOffset());
+                const auto newValue = numRows + currentOffset;
+
+                // Update the Core's viewport position, and raise a
+                // ScrollPositionChanged event to update the scrollbar
+                _updateScrollbar(newValue);
+
+                // Use this point as our new scroll anchor.
+                _touchAnchor = newTouchPoint;
+            }
+        }
+    }
+
+    void ControlInteractivity::PointerReleased(TerminalInput::MouseButtonState buttonState,
+                                               const unsigned int pointerUpdateKind,
+                                               const ::Microsoft::Terminal::Core::ControlKeyStates modifiers,
+                                               const bool /*focused*/,
+                                               const til::point terminalPosition)
+    {
+        // Short-circuit isReadOnly check to avoid warning dialog
+        if (!_core->IsInReadOnlyMode() && _canSendVTMouseInput(modifiers))
+        {
+            _trySendMouseEvent(pointerUpdateKind, buttonState, modifiers, terminalPosition);
+            return;
+        }
+
+        // Only a left click release when copy on select is active should perform a copy.
+        // Right clicks and middle clicks should not need to do anything when released.
+        const bool isLeftMouseRelease = pointerUpdateKind == WM_LBUTTONUP;
+
+        if (_core->CopyOnSelect() &&
+            isLeftMouseRelease &&
+            _selectionNeedsToBeCopied)
+        {
+            CopySelectionToClipboard(false, nullptr);
+        }
+
+        _singleClickTouchdownPos = std::nullopt;
+        _singleClickTouchdownTerminalPos = std::nullopt;
+    }
+
+    void ControlInteractivity::TouchReleased()
+    {
+        _touchAnchor = std::nullopt;
+    }
+
+    // Method Description:
+    // - Send this particular mouse event to the terminal.
+    //   See Terminal::SendMouseEvent for more information.
+    // Arguments:
+    // - point: the PointerPoint object representing a mouse event from our XAML input handler
+    bool ControlInteractivity::_trySendMouseEvent(const unsigned int updateKind,
+                                                  const TerminalInput::MouseButtonState buttonState,
+                                                  const ::Microsoft::Terminal::Core::ControlKeyStates modifiers,
+                                                  const til::point terminalPosition)
+    {
+        return _core->SendMouseEvent(terminalPosition, updateKind, modifiers, 0, buttonState);

```

Comment:
since Interactivity is only a partial cover for Core, it might be helpful to understand why we don't just call core->SendMouseEvent in interactivity

(i think that Control should not be sending mouse events directly to core, so this helper would _always_ be needed... but i think that control should not be handling its own mouse input, so this helper will never be needed (as all mouse event comes in through Interactivity::DoMouseThing))

Reply:
I guess what I'm saying is:

All mouse event comes in through Interactivity, and Interactivity doesn't seem to use these helpers.

NOBODY else should be sending raw VT mouse input into Core.

Reply:
Wait so is the gist, "yank these helpers, they add nothing. Just replace them with calls to `core->SendMouseEvent`"? That makes sense to me. Ping me if that's _not_ what you mean here